### PR TITLE
Improve coverage and tests.

### PR DESCRIPTION
**Pull request by tobiasraabe**
### Current behavior

What is the current state?

### Desired behavior

What do you want to achieve?

### Solution / Implementation

What is your solution?

### Todo

- [ ] Review whether the documentation needs to be updated.
- [ ] Document PR in CHANGES.rst.

### CODE REVIEW

File: tests/test_shared.py [hunk 1]
```diff
@@ -26,3 +32,43 @@ def test_factorize_assortative_variables(initial_states, assort_by, expected):
...
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "choices, probs, seed, expectation, expected",
+    [
+        (3, [0, 0, 1], None, does_not_raise(), 2),
+        ([0, 1, 2], [0, 1, 0], None, does_not_raise(), 1),
+        (np.arange(3), [1, 0, 0], None, does_not_raise(), 0),
+        (3, None, 0, does_not_raise(), 1),
```
Comment (roecla) [inline tests/test_shared.py:54]: Why is 1 the expected result here?

**Review by roecla — CHANGES_REQUESTED** [review]
Very nice. :clap:  

However, I get an error when running the tests: 

```python

====================================== FAILURES ======================================
__________________________ test_boolean_choice[0.7-0-True] ___________________________

x = 0.7, seed = 0, expected = True

    @pytest.mark.unit
    @pytest.mark.parametrize(
        "x, seed, expected", [(0, None, False), (1, None, True), (0.7, 0, True)]
    )
    def test_boolean_choice(x, seed, expected):
        np.random.seed(seed)
        result = boolean_choice(x)
>       assert result is expected
E       assert False is True

tests/test_shared.py:44: AssertionError
```

I've freshly installed the sid environment and there is a seed. Any ideas where this could come from?

File: tests/test_shared.py (hunk 1 above)
Comment (tobiasraabe) [reply tests/test_shared.py:54]: I set the seed to zero which results in 1. It is not a deterministic edge case if this was your question.

File: src/sid/matching_probabilities.py [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 
 
 def create_cumulative_group_transition_probabilities(
     states, assort_by, params, model_name, groups
 ):
     """Create a transition matrix for groups.
 
+    If the model has no ``assort_by`` variables,
+    :func:`sid.shared.factorize_assortative_variables` create a group column with a
+    single group.
```
[the remark this answers is not in the document]
Comment (tobiasraabe) [reply src/sid/matching_probabilities.py:20]: Hope I clarified it.

**Comment by tobiasraabe** [comment]
Regarding the general comment, I am stupid. I did not look at the function which is compiled with Numba. Obviously, setting the usual numpy seed does not work. Since we also do not want to set the seed inside the function, I removed the test case.

**Review by janosg — APPROVED** [review]

**Review by janosg — APPROVED** [review]
